pkg/scc/systeminfo/offline/offline.go

```diff
@@ -0,0 +1,64 @@
+package offline
+
+import (
+	"errors"
+	"fmt"
+	"github.com/SUSE/connect-ng/pkg/registration"
+	"github.com/rancher/rancher/pkg/scc/systeminfo"
+	"time"
+)
+
+type CertificateValidator struct {
+	offlineCert        *registration.OfflineCertificate
+	systemInfoExporter *systeminfo.InfoExporter
+}
+
+func New(offlineCert *registration.OfflineCertificate, systemInfoExporter *systeminfo.InfoExporter) *CertificateValidator {
+	return &CertificateValidator{
+		offlineCert:        offlineCert,
+		systemInfoExporter: systemInfoExporter,
+	}
+}
+
+func (cv *CertificateValidator) ValidateCertificate() error {
+	certIsValid, validateErr := cv.offlineCert.IsValid()
+	if validateErr != nil {
+		return validateErr
+	}
+
+	if !certIsValid {
+		return errors.New("certificate is not valid")
+	}
+
+	// TODO: does offline need users to define RegCode for this?
+	// matchesRegCode, regCodeErr := cv.offlineCert.RegcodeMatches("")
+
+	rancherUUID := cv.systemInfoExporter.RancherUuid()
+	matchesRancherUUID, uidErr := cv.offlineCert.UUIDMatches(rancherUUID.String())
+	if uidErr != nil {
+		return fmt.Errorf("failed to validate Rancher UUID: %v", uidErr)
+	}
+
+	if !matchesRancherUUID {
```

Comment:
This check needs a special case so we can allow for RGS: either we always accept the UUID if the one in the cert is set to 0x0, or we add a flag to the cert that skips UUID validation (might be cleaner since it doesn't overload the UUID values).

Reply:
Add this in newest implementation.